include/oneapi/tbb/parallel_for.h

```diff
@@ -310,7 +305,7 @@ void parallel_for_impl(Index first, Index last, Index step, const Function& f, P
         throw_exception(exception_id::nonpositive_step); // throws std::invalid_argument
     else if (first < last) {
         // Above "else" avoids "potential divide by zero" warning on some platforms
-        Index end = (last - first - Index(1)) / step + Index(1);
+        Index end = Index(last - first - 1ul) / step + Index(1);
```

Comment:
This seems unrelated to std::invoke.

Reply:
Yes, it is not related for `std::invoke`, but I failed to implement custom `ParallelForIndex` as it defined in the spec to make this line work. I faced a lot of problems with the operation priority, conversions and wrong return values for mathematical operations. This workaround can be considered as partial fix for the implementation to better match the spec.